tests/acceptance/features/apiSharingNg/deletePermissions.feature

```diff
@@ -0,0 +1,127 @@
+      | Manager     | file          | textfile.txt  |
+      | Viewer      | folder        | FolderToShare |
+      | Editor      | folder        | FolderToShare |
+      | Co Owner    | folder        | FolderToShare |
+      | Uploader    | folder        | FolderToShare |
+      | Manager     | folder        | FolderToShare |
+
+
+  Scenario Outline: user removes access to a resource in a group share
+    Given group "group1" has been created
+    And user "Brian" has been added to group "group1"
+    And user "Alice" has been added to group "group1"
+    And user "Alice" has created folder "FolderToShare"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "textfile.txt"
+    And user "Alice" has sent the following share invitation:
+      | resourceType | <resource-type> |
+      | resource     | <path>          |
+      | space        | Personal        |
+      | sharee       | group1          |
+      | shareType    | group           |
+      | role         | <role>          |
+    When user "Alice" removes the share permission of group "group1" from <resource-type> "<path>" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    And for user "Brian" the space "Shares" should not contain these entries:
+      | <path> |
+    Examples:
+      | role        | resource-type | path          |
+      | Viewer      | file          | textfile.txt  |
+      | File Editor | file          | textfile.txt  |
+      | Co Owner    | file          | textfile.txt  |
+      | Manager     | file          | textfile.txt  |
+      | Viewer      | folder        | FolderToShare |
+      | Editor      | folder        | FolderToShare |
+      | Co Owner    | folder        | FolderToShare |
+      | Uploader    | folder        | FolderToShare |
+      | Manager     | folder        | FolderToShare |
+
+
+  Scenario Outline: user removes access to a resource inside of a project space in group share
+    Given the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "NewSpace" with the default quota using the Graph API
+    And user "Alice" has created a folder "FolderToShare" in space "NewSpace"
+    And user "Alice" has uploaded a file inside space "NewSpace" with content "some content" to "textfile.txt"
+    And group "group1" has been created
+    And user "Brian" has been added to group "group1"
+    And user "Alice" has been added to group "group1"
+    And user "Alice" has sent the following share invitation:
+      | resourceType | <resource-type> |
+      | resource     | <path>          |
+      | space        | NewSpace        |
+      | sharee       | group1          |
+      | shareType    | group           |
+      | role         | <role>          |
+    When user "Alice" removes the share permission of group "group1" from <resource-type> "<path>" of space "NewSpace" using the Graph API
```

Comment:
we have 2 options invite people (role and permission): 
```
When user "Alice" sends the following share invitation using the Graph API:
      | resourceType | folder        |
      | role     | <role> |
```

and 

```
When user "Alice" sends the following share invitation using the Graph API:
      | resourceType | folder        |
      | permission     | <permission> |
```

In your case you invite people using role. Probably in this case better to use something like: 
`When user "Alice" removes the role "<role>" of group "group1" from <resource-type> "<path>" of space "NewSpace" using the Graph API` 

and getting roleId using `$roleId = self::getRoleIdByName($role)` (ids are hardcoded, we don't need send `getPermissionsList()`every time)


But question(I just no clue): How can we delete permissions? if we send invite using permissions instead of role

Reply:
If I'm not mistaken the `permission-id` and `role-id` are different. I started implementing as you suggested but in swagger  the endpoint is something like this `/v1beta1/drives/{drive-id}/items/{item-id}/permissions/{perm-id}`  https://owncloud.dev/libre-graph-api/#/drives.permissions/DeletePermission

What I'm doing is I'm taking the value part of the `listPermission` endpoint
```json
"value": [
    {
      "grantedToV2": {
        "user": {
          "displayName": "Albert Einstein",
          "id": "4c510ada-c86b-4815-8820-42cdf82c3d51"
        }
      },
      "id": "df442f0d-9fbb-4ce6-9ea3-95db836e4f3a:602db7b5-a022-4ea2-b171-bad88e9ff780:6a736352-d918-4d4c-b56e-04fa7ba76859",
      "roles": [
        "b1e2218d-eef8-4d4c-b82d-0f1a1b48f3b5"
      ]
    },
    {
      "grantedToV2": {
        "user": {
          "displayName": "Marie Skłodowska Curie",
          "id": "f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c"
        }
      },
      "id": "df442f0d-9fbb-4ce6-9ea3-95db836e4f3a:602db7b5-a022-4ea2-b171-bad88e9ff780:2177ca6f-631f-418a-bbb4-137d55da9b03",
      "roles": [
        "b1e2218d-eef8-4d4c-b82d-0f1a1b48f3b5"
      ]
    },
    {
      "hasPassword": true,
      "id": "onliEyuWVxuFDFt",
      "link": {
        "@libre.graph.displayName": "Link",
        "@libre.graph.quickLink": true,
        "preventsDownload": false,
        "type": "view",
        "webUrl": "https://localhost:9200/s/moUJWwwjcNtexSw"
      }
    }
  ]
```

On each object, there's and id and I'm taking that Id which I think is the `perm-id` as the delete request works with it. But not with the `roleId` but one thing that I just realized is that maybe the `perm-Id` is shareId(need to confirm)? Maybe I can store it from the invite response and use it for this request.

Is my understanding of this correct or am I doing this wrong?

Reply:
Sorry Swikriti. I found my mistake. naming is really confused
in request `DELETE /{drive-id}/items/{item-id}/permissions/{perm-id}` we are using permissions/{perm-id} what actually is shareId

<img width="1231" alt="Screenshot 2024-01-05 at 10 31 07" src="https://github.com/owncloud/ocis/assets/84779829/52a613e4-1441-418f-84b4-8ae775b20f95">

Reply:
My suggestion to changing naming https://github.com/owncloud/ocis/pull/8140